### PR TITLE
[ADD] Add shopinvader product stock + shopinvader algolia product stock

### PR DESCRIPTION
**Pull request by acsonefho**
This MR depends on the MR akretion/connector-search-engine/pull/24 + 26

Actually, builds failed because MR 24 and 26 (of connector-search-engine) are not merged and these MR are not into dependencies

### CODE REVIEW

File: shopinvader_algolia_stock/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Akretion (http://www.akretion.com)
+# Copyright 2018 ACSONE SA/NV
+# Sébastien BEAU <sebastien.beau@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+{
+    "name": "Connector Search Engine Algolia stock",
+    "summary": "Shopinvader stock algolia",
+    'description': """
+        Connector search engine for Algolia Stock""",
```
Comment (lmignon) [inline shopinvader_algolia_stock/__manifest__.py]: @acsonefho The description should be more meaningful.  'This module adds the stock levele of your products to the informations exported to Aalgolia'

File: shopinvader_stock/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Akretion (http://www.akretion.com)
+# Copyright 2018 ACSONE SA/NV
+# Sébastien BEAU <sebastien.beau@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+{
+    "name": "Connector Search Engine stock",
+    "summary": "Connector Search Engine stock",
+    'description': """
+        Connector search engine for Stock""",
```
Comment (lmignon) [inline shopinvader_stock/__manifest__.py]: @acsonefho same comment as above

File: shopinvader_stock/models/stock_move.py [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+        """
+        Get the mapper to use.
+        Return None to have the default mapper
+        :return: str
+        """
+        return None
+
+    @api.multi
+    @job(default_channel='root.search_engine')
+    def _product_stock_update_all(self):
```
Comment (lmignon) [inline shopinvader_stock/models/stock_move.py]: @acsonefho These following 3 methods should be placed into product.product...

File: shopinvader_algolia_product_stock/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        'shopinvader_algolia',
+    ],
+    "data": [
+        'views/se_backend_algolia.xml',
+    ],
+    'external_dependencies': {
+        'python': [
+            'openupgradelib',
+        ],
+    },
```
Comment (Cedric-Pigeon) [inline shopinvader_algolia_product_stock/__manifest__.py:31]: 'pre_init_hook': 'rename_module' is missing

File: shopinvader_product_stock/__manifest__.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    ],
+    "data": [
+        'views/se_backend.xml',
+    ],
+    'external_dependencies': {
+        'python': [
+            'openupgradelib',
+        ],
+    },
+    'pre_init_hook': 'rename_module',
```
Comment (lmignon) [inline shopinvader_product_stock/__manifest__.py]: This name is not conform to OCA guidelines

File: shopinvader_product_stock/models/product_product.py [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+        :param products: product.product recordset
+        :return: shopinvader.variant recordset
+        """
+        precision = self.env['decimal.precision'].precision_get(
+            'Product Unit of Measure')
+        products = products.filtered(
+            lambda p: p.is_shopinvader_binded).with_prefetch(self._prefetch)
+        # Get every shopinvader variant where the backend has a json key to
+        # export
+        shopinvader_variants = products.shopinvader_bind_ids.filtered(
```
Comment (rousseldenis) [inline shopinvader_product_stock/models/product_product.py:80]: @acsonefho products can be multi, so it raises error. Use mapped on shopinvader_bind_ids

File: shopinvader_product_stock/models/product_product.py [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+        index.
+        This result will be given to another job (1 job/index). So in case
+        of failure, only the export of 1 index should be re-launched.
+        :return:
+        """
+        if not self._job_stock_can_start():
+            message = _("Another related queue job is running...")
+            raise exceptions.MissingError(message)
+        products = self
+        shopinvader_variants = self._get_shopinvader_variants_from_product(
```
Comment (rousseldenis) [inline shopinvader_product_stock/models/product_product.py:46]: @acsonefho Maybe creating a 'shopinvader_variants' computed field on product.product ?

File: shopinvader_product_stock/models/se_backend.py [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+# Copyright 2018 ACSONE SA/NV
+# Sébastien BEAU <sebastien.beau@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import api, exceptions, fields, models, _
+
+
+class SeBackend(models.Model):
+    _inherit = 'se.backend'
+
+    product_stock_field_id = fields.Many2one(
```
Comment (rousseldenis) [inline shopinvader_product_stock/models/se_backend.py:12]: Is this working without being administrator (in view)?

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Some remarks (one blocking)

**Comment by sebastienbeau** [comment]
I have done a PR that includ this one with a refactoring in order to:
- use the existing exporter for defining the key in the export
- support specific warehouse for stock level (also support multi warehouse)
- export in algolia only if stock level have change, as export is now done in two step (first compute json independently, then export it in batch)
The new PR is here : https://github.com/akretion/odoo-shopinvader/pull/182

**Comment by sebastienbeau** [comment]
This have been merged here with a refactoring : https://github.com/akretion/odoo-shopinvader/pull/182

**Pull request closed without merging**